src/charm.py

```diff
@@ -45,24 +45,26 @@ def _on_upgrade_charm(self, _) -> None:
         self.unit.status = ops.MaintenanceStatus("Installing packages")
         apt.add_package(["libxml2"], update_cache=True)
 
-    def _on_relation_created(self, event: ops.RelationCreatedEvent) -> None:
-        """Handle a change to the saml relation.
-
-        Args:
-            event: Event triggering the relation-created hook for the relation.
-        """
+    def _on_relation_created(self, _) -> None:
+        """Handle a change to the saml relation."""
         if not self.model.unit.is_leader():
             return
-        self.saml.update_relation_data(event.relation, self.get_saml_data())
+        # A new charm will be instantiated hance, the information will be fetched again.
```

Comment:
is `hance` spelled correctly?

Reply:
I'm not following the point regarding a new charm will be instantiated?

Reply:
Fixed the typo. Whenever a new event arrives, the charms are always instantiated again. This means that the cached properties might change